services/codeclimate/codeclimate-analysis.tester.js

```diff
@@ -43,29 +37,29 @@ t.create('maintainability letter')
 t.create('maintainability letter for non-existent repo')
   .get('/maintainability/unknown/unknown.json')
   .expectBadge({
-    label: 'maintainability',
-    message: 'not found',
+    label: 'analysis',
+    message: 'repo not found',
   })
 
 t.create('maintainability letter for repo without snapshots')
   .get('/maintainability/kabisaict/flow.json')
   .expectBadge({
-    label: 'maintainability',
-    message: 'unknown',
+    label: 'analysis',
+    message: 'snapshot not found',
   })
 
 t.create('malformed response for outer user repos query')
   .get('/maintainability/angular/angular.js.json')
   .intercept(nock =>
     nock('https://api.codeclimate.com')
-      .get('/v1/repos?github_slug=angular/angular.js')
+      .get('/v1/repos?github_slug=angular%2Fangular.js')
       .reply(200, {
         data: [{}], // No relationships in the list of data elements.
```

Comment:
I keep going back and forth on this test. What sort of scenarios in real life would result in us getting this API response? If the `/v1/repos` call includes a repo slug that doesn't exist, data is an empty array and we'd display a `repo not found` badge.

I can't picture what would cause this response but if we know of one then maybe we could update this comment to call that out for future observers? Something like `// No relationships in the list of data elements. This happens when the repo...`

Reply:
Just realized your below comment covered this test too 😄

Reply:
Having thought about this a bit more, I realize the bugs these tests caught were a result of not correctly detecting the not found case (which the schema explicitly lets in). Implementing #2632 would eliminate that error surface.

Reply:
I still can't wrap my head around what would make the scenario in this first test actually occur (I thought not found would always be `data: []`), but happy to defer to your judgement on keeping these i👍